### PR TITLE
EZP 25711: Fix call for subtree alias generation 

### PR DESCRIPTION
**Pull request by galileo**
Fixes: https://jira.ez.no/browse/EZP-25711

> Status: Ready for review

I have fixed the call to `locationCopied` from locationService. 

I think the problem source was the bad doc block for params in `eZ/Publish/SPI/Persistence/Content/UrlAlias/Handler.php` and this forced the missuse of aliasHelper method.

### CODE REVIEW

File: eZ/Publish/Core/Persistence/Cache/UrlAliasHandler.php [hunk 1]
```diff
@@ -251,26 +251,33 @@ public function locationMoved($locationId, $oldParentId, $newParentId)
         );
 
         $return = $this->persistenceHandler->urlAliasHandler()->locationMoved($locationId, $oldParentId, $newParentId);
+        $this->cache->clear('urlAlias');
 
-        $this->cache->clear('urlAlias');//TIMBER! (Will have to load url aliases for location to be able to clear specific entries)
```
Comment (andrerom) [inline eZ/Publish/Core/Persistence/Cache/UrlAliasHandler.php]: Can leave it as is

File: eZ/Publish/Core/Persistence/Cache/UrlAliasHandler.php [hunk 2]
```diff
@@ -251,26 +251,33 @@ public function locationMoved($locationId, $oldParentId, $newParentId)
...
             )
         );
 
-        return $this->persistenceHandler->urlAliasHandler()->locationCopied($locationId, $oldParentId, $newParentId);
+        $return = $this->persistenceHandler->urlAliasHandler()->locationCopied(
+            $locationId,
+            $newLocationId,
+            $newParentId
+        );
+        $this->cache->clear('urlAlias');
```
Comment (andrerom) [inline eZ/Publish/Core/Persistence/Cache/UrlAliasHandler.php]: As mentioned, think this one can be set as `$this->cache->clear('urlAlias', 'url');` given we only care about possible negative matches when creating new entries.

Comment (galileo) [inline eZ/Publish/Core/Persistence/Cache/UrlAliasHandler.php]: Sure you are right, not much but always a performance boost.

File: eZ/Publish/Core/Persistence/Cache/UrlAliasHandler.php (hunk 1 above)
Comment (galileo) [inline eZ/Publish/Core/Persistence/Cache/UrlAliasHandler.php]: sure

**Comment by andrerom** [comment]
+1

**Comment by andrerom** [comment]
behat failure is unrelated and fixed separately, merging.

File: eZ/Publish/API/Repository/Tests/LocationServiceTest.php [hunk 3]
```diff
@@ -1484,6 +1484,46 @@ public function testCopySubtree()
...
+
+        $expectedSubItemAliases = [
+            '/Design/Plain-site/Media/Multimedia',
+            '/Design/Plain-site/Media/Images',
+            '/Design/Plain-site/Media/Files',
+        ];
+
+        $this->assertAliasesBeforeCopy($urlAliasService, $expectedSubItemAliases);
+
+        // Copy location "Media" to "Design"
```
Comment (bdunogier) [inline eZ/Publish/API/Repository/Tests/LocationServiceTest.php]: A bit late, and not omportant, but if you had named `$locationToCopy` and `$newParentLocation` after what they are instead, the comment would not have been necessary.

`$locationService->copySubtree($mediaLocation, $demoDesignLocation)`

**Pull request merged**